### ISSUE TITLE
Issue with DB event listener connector

### ISSUE THREAD

**Issue by abeykoon** [wso2/micro-integrator#1123]
**Description:**

**Affected Product Version:**

Connector version : 1.0.3 

**OS, DB, other environment details and versions:**    

EI 6.6.0

**Steps to reproduce:**

Please follow doc [1]. After doing one invocation, it keeps triggering events for the same record. 

```
<?xml version="1.0" encoding="UTF-8"?>
<inboundEndpoint class="org.wso2.carbon.inbound.poll.dbeventlistener.DBEventPollingConsumer" name="dbEventIE" onError="DBEventFaultSeq" sequence="dbEventProcessSeq" suspend="false" xmlns="http://ws.apache.org/ns/synapse">
    <parameters>
        <parameter name="inbound.behavior">polling</parameter>
        <parameter name="interval">10000</parameter>
        <parameter name="sequential">true</parameter>
        <parameter name="coordination">true</parameter>
        <parameter name="driverName">com.mysql.jdbc.Driver</parameter>
        <parameter name="url">jdbc:mysql://localhost/test</parameter>
        <parameter name="username">root</parameter>
        <parameter name="password">root</parameter>
        <parameter name="tableName">CDC_CUSTOM</parameter>
        <parameter name="filteringCriteria">byLastUpdatedTimestampColumn</parameter>
        <parameter name="filteringColumnName">AMOUNT</parameter>
        <parameter name="primaryKey">ID</parameter>
        <parameter name="connectionValidationQuery">SELECT 1</parameter>
        <parameter name="registryPath">dbEventIE/timestamp</parameter>
        <parameter name="class">org.wso2.carbon.inbound.poll.dbeventlistener.DBEventPollingConsumer</parameter>
    </parameters>
</inboundEndpoint>
```

Log: 

```
2020-01-24 14:38:40,119]  INFO {org.apache.synapse.mediators.builtin.LogMediator} - To: , MessageID: urn:uuid:22781C4AC683DE6AB91579856920134, Direction: request, Envelope: <?xml version='1.0' encoding='utf-8'?><soapenv:Envelope xmlns:soapenv="http://www.w3.org/2003/05/soap-envelope"><soapenv:Body><Record><ID>1</ID><NAME>john</NAME><ADDRESS>22/3, Tottenham Court, London</ADDRESS><AMOUNT>6000</AMOUNT></Record></soapenv:Body></soapenv:Envelope>
[2020-01-24 14:38:50,116]  INFO {org.apache.synapse.mediators.builtin.LogMediator} - To: , MessageID: urn:uuid:22781C4AC683DE6AB91579856930133, Direction: request, Envelope: <?xml version='1.0' encoding='utf-8'?><soapenv:Envelope xmlns:soapenv="http://www.w3.org/2003/05/soap-envelope"><soapenv:Body><Record><ID>1</ID><NAME>john</NAME><ADDRESS>22/3, Tottenham Court, London</ADDRESS><AMOUNT>6000</AMOUNT></Record></soapenv:Body></soapenv:Envelope>
[2020-01-24 14:39:00,114]  INFO {org.apache.synapse.mediators.builtin.LogMediator} - To: , MessageID: urn:uuid:22781C4AC683DE6AB91579856940132, Direction: request, Envelope: <?xml version='1.0' encoding='utf-8'?><soapenv:Envelope xmlns:soapenv="http://www.w3.org/2003/05/soap-envelope"><soapenv:Body><Record><ID>1</ID><NAME>john</NAME><ADDRESS>22/3, Tottenham Court, London</ADDRESS><AMOUNT>6000</AMOUNT></Record></soapenv:Body></soapenv:Envelope>
[2020-01-24 14:39:10,113]  INFO {org.apache.synapse.mediators.builtin.LogMediator} - To: , MessageID: urn:uuid:22781C4AC683DE6AB91579856950132, Direction: request, Envelope: <?xml version='1.0' encoding='utf-8'?><soapenv:Envelope xmlns:soapenv="http://www.w3.org/2003/05/soap-envelope"><soapenv:Body><Record><ID>1</ID><NAME>john</NAME><ADDRESS>22/3, Tottenham Court, London</ADDRESS><AMOUNT>6000</AMOUNT></Record></soapenv:Body></soapenv:Envelope>
[2020-01-24 14:39:20,118]  INFO {org.apache.synapse.mediators.builtin.LogMediator} - To: , MessageID: urn:uuid:22781C4AC683DE6AB91579856960137, Direction: request, Envelope: <?xml version='1.0' encoding='utf-8'?><soapenv:Envelope xmlns:soapenv="http://www.w3.org/2003/05/soap-envelope"><soapenv:Body><Record><ID>1</ID><NAME>john</NAME><ADDRESS>22/3, Tottenham Court, London</ADDRESS><AMOUNT>6000</AMOUNT></Record></soapenv:Body></soapenv:Envelope>
[2020-01-24 14:39:30,119]  INFO {org.apache.synapse.mediators.builtin.LogMediator} - To: , MessageID: urn:uuid:22781C4AC683DE6AB91579856970139, Direction: request, Envelope: <?xml version='1.0' encoding='utf-8'?><soapenv:Envelope xmlns:soapenv="http://www.w3.org/2003/05/soap-envelope"><soapenv:Body><Record><ID>1</ID><NAME>john</NAME><ADDRESS>22/3, Tottenham Court, London</ADDRESS><AMOUNT>6000</AMOUNT></Record></soapenv:Body></soapenv:Envelope>
[2020-01-24 14:39:40,113]  INFO {org.apache.synapse.mediators.builtin.LogMediator} - To: , MessageID: urn:uuid:22781C4AC683DE6AB91579856980135, Direction: request, Envelope: <?xml version='1.0' encoding='utf-8'?><soapenv:Envelope xmlns:soapenv="http://www.w3.org/2003/05/soap-envelope"><soapenv:Body><Record><ID>1</ID><NAME>john</NAME><ADDRESS>22/3, Tottenham Court, London</ADDRESS><AMOUNT>6000</AMOUNT></Record></soapenv:Body></soapenv:Envelope>
[2020-01-24 14:39:50,114]  INFO {org.apache.synapse.mediators.builtin.LogMediator} - To: , MessageID: urn:uuid:22781C4AC683DE6AB91579856990137, Direction: request, Envelope: <?xml version='1.0' encoding='utf-8'?><soapenv:Envelope xmlns:soapenv="http://www.w3.org/2003/05/soap-envelope"><soapenv:Body><Record><ID>1</ID><NAME>john</NAME><ADDRESS>22/3, Tottenham Court, London</ADDRESS><AMOUNT>6000</AMOUNT></Record></soapenv:Body></soapenv:Envelope>
[2020-01-24 14:40:00,115]  INFO {org.apache.synapse.mediators.builtin.LogMediator} - To: , MessageID: urn:uuid:22781C4AC683DE6AB91579857000139, Direction: request, Envelope: <?xml version='1.0' encoding='utf-8'?><soapenv:Envelope xmlns:soapenv="http://www.w3.org/2003/05/soap-envelope"><soapenv:Body><Record><ID>1</ID><NAME>john</NAME><ADDRESS>22/3, Tottenham Court, London</ADDRESS><AMOUNT>6000</AMOUNT></Record></soapenv:Body></soapenv:Envelope>
[2020-01-24 14:40:10,117]  INFO {org.apache.synapse.mediators.builtin.LogMediator} - To: , MessageID: urn:uuid:22781C4AC683DE6AB91579857010142, Direction: request, Envelope: <?xml version='1.0' encoding='utf-8'?><soapenv:Envelope xmlns:soapenv="http://www.w3.org/2003/05/soap-envelope"><soapenv:Body><Record><ID>1</ID><NAME>john</NAME><ADDRESS>22/3, Tottenham Court, London</ADDRESS><AMOUNT>6000</AMOUNT></Record></soapenv:Body></soapenv:Envelope>
[2020-01-24 14:40:20,112]  INFO {org.apache.synapse.mediators.builtin.LogMediator} - To: , MessageID: urn:uuid:22781C4AC683DE6AB91579857020138, Direction: request, Envelope: <?xml version='1.0' encoding='utf-8'?><soapenv:Envelope xmlns:soapenv="http://www.w3.org/2003/05/soap-envelope"><soapenv:Body><Record><ID>1</ID><NAME>john</NAME><ADDRESS>22/3, Tottenham Court, London</ADDRESS><AMOUNT>6000</AMOUNT></Record></soapenv:Body></soapenv:Envelope>
[2020-01-24 14:40:30,115]  INFO {org.apache.synapse.mediators.builtin.LogMediator} - To: , MessageID: urn:uuid:22781C4AC683DE6AB91579857030141, Direction: request, Envelope: <?xml version='1.0' encoding='utf-8'?><soapenv:Envelope xmlns:soapenv="http://www.w3.org/2003/05/soap-envelope"><soapenv:Body><Record><ID>1</ID><NAME>john</NAME><ADDRESS>22/3, Tottenham Court, London</ADDRESS><AMOUNT>6000</AMOUNT></Record></soapenv:Body></soapenv:Envelope>
[2020-01-24 14:40:40,113]  INFO {org.apache.synapse.mediators.builtin.LogMediator} - To: , MessageID: urn:uuid:22781C4AC683DE6AB91579857040141, Direction: request, Envelope: <?xml version='1.0' encoding='utf-8'?><soapenv:Envelope xmlns:soapenv="http://www.w3.org/2003/05/soap-envelope"><soapenv:Body><Record><ID>1</ID><NAME>john</NAME><ADDRESS>22/3, Tottenham Court, London</ADDRESS><AMOUNT>6000</AMOUNT></Record></soapenv:Body></soapenv:Envelope>
[2020-01-24 14:40:50,117]  INFO {org.apache.synapse.mediators.builtin.LogMediator} - To: , MessageID: urn:uuid:22781C4AC683DE6AB91579857050145, Direction: request, Envelope: <?xml version='1.0' encoding='utf-8'?><soapenv:Envelope xmlns:soapenv="http://www.w3.org/2003/05/soap-envelope"><soapenv:Body><Record><ID>1</ID><NAME>john</NAME><ADDRESS>22/3, Tottenham Court, London</ADDRESS><AMOUNT>6000</AMOUNT></Record></soapenv:Body></soapenv:Envelope>
```
[1]. https://docs.wso2.com/display/ESBCONNECTORS/Working+with+the+DB+Event+Listener

**Related Issues:**
N/A

**Comment by sdkottegoda** [comment]
@abeykoon , did you test the connector against EI-6.6.0? The issues reported here should be the ones tested against Micro Integrator

**Comment by sdkottegoda** [comment]
Also, did you not face the issue mentioned at https://github.com/wso2/micro-integrator/issues/1115 while deploying the inbound?

**Comment by abeykoon** [comment]
The connector is not compatible with MI. 
Raised issue to track : https://github.com/wso2/micro-integrator/issues/1238

**Comment by abeykoon** [comment]
This issue is fixed with the DB event listener v1.0.4
